Fleece/Core/JSONConverter.cc

```diff
@@ -32,7 +32,7 @@ namespace fleece { namespace impl {
 
     JSONConverter::JSONConverter(Encoder &e) noexcept
     :_encoder(e),
-     _jsn(jsonsl_new(50)),      // never returns nullptr, according to source code
+     _jsn(jsonsl_new(UINT_MAX)),      // never returns nullptr, according to source code
```

Comment:
Did you happen to notice what this does inside jsonsl? From jsonsl.c line 111:
```c
            calloc(1, sizeof (*jsn) +
                    ( (nlevels-1) * sizeof (struct jsonsl_state_st) )
```
This patch will cause jsonsl to allocate a huge amount of memory -- 4 billion times the size of a struct that's about 50 bytes large, i.e. around 200GB.

The OS kernel may or may not commit all that address space. If it doesn't, things are OK. If it does, that's an immediate out-of-memory error. I suspect iOS in particular will have the latter behavior since it doesn't fully use virtual memory.

Reply:
Thank you! It's an honor to get told by the creator of the ubiquitous chat bubble 🙇🏻 

I apologize for the oversight- it has been a long time since I have used c++ and had made some assumptions about the implementation. This is definitely a no no.